core-bundle/src/Resources/public/autofocus.js

```diff
@@ -0,0 +1,37 @@
+/*!
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+window.addEventListener('DOMContentLoaded', function () {
+    const edit = document.querySelector('#main .tl_formbody_edit');
+    if (!edit) return;
+
+    // Copy from MooTools
+    function isHidden (el) {
+        const w = el.offsetWidth, h = el.offsetHeight,
+            force = /^tr$/i.test(el.tagName);
+        return (w===0 && h===0 && !force) ? true : (w!==0 && h!==0 && !force) ? false : el.style.display === 'none';
+    }
+
+    const inputs = edit.querySelectorAll('input, textarea');
+
+    for (let i = 0; i < inputs.length; i++) {
+        const input = inputs[i];
+
+        if (
+            !input.getAttribute('disabled')
+            && !input.getAttribute('readonly')
```

Comment:
```suggestion
            !input.disabled
            && !input.readonly
```

The mootools `get()` method is not synonymous with `getAttribute`.

Reply:
According to the Mootols code, it actually is pretty synonymous. Even though `getAttribute` and DOM properties are _almost always_ synonymous (they are in this case), https://javascript.info/dom-attributes-and-properties suggests to use the DOM property since it would probably also be boolean in this case. 👍

Reply:
Side note: using `getAttribute()` for boolean attributes gives you the empty string for the `true` value, need to use `hasAttribute()` instead then to get a boolean.
But 👍 for using DOM properties instead.

Reply:
Changed in b4b4b733ab915ced966e20ea3f073d28bc27d52f